staging/src/k8s.io/apimachinery/pkg/watch/mux.go

```diff
@@ -96,10 +94,15 @@ func (obj functionFakeRuntimeObject) DeepCopyObject() runtime.Object {
 // The purpose of this terrible hack is so that watchers added after an event
 // won't ever see that event, and will always see any event after they are
 // added.
-func (b *Broadcaster) blockQueue(f func()) {
+func (m *Broadcaster) blockQueue(f func()) {
+	select {
+	case <-m.stopped:
```

Comment:
previously, sending after shutdown would result in a write to a closed channel, right?  Can you make a unit test to prove that it used to panic?

Reply:
blockQueue is called by Watch, WatchWithPrefix, stopWatching, Shutdown. The Action method sends data directly to incoming chan instead of calling blockQueue. There is a subtle bug here. If the Broadcaster has been closed, calling Watch will get a watch.Interface with a value of nil instead of Panic as before. If I call panic here, calling Watcher's Stop method after closing the Broadcaster will cause Panic. In order to maintain the consistency of the interface, in the Watch && WatchWithPrefix method, determine whether broadcasterWatcher is nil to decide whether to call panic.